From dee3a84028ddded9ab129934dd79aca70d3e9537 Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Tue, 20 May 2014 09:36:43 +0800
Subject: [PATCH] gui: disable unsupported options

Disable the following unsupported options from the GUI:
* Add repo
* Edit repo

And make the repo's "toggle button" gray (insenstive), we still need
them to be visible to see which repos will be used.

We will revert this patch when we can support the network installs.

Upstream-Status: Inappropriate [WR Specific]

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 data/ui/tasksel.glade     | 2 +-
 pyanaconda/iw/task_gui.py | 9 +++++++--
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/data/ui/tasksel.glade b/data/ui/tasksel.glade
index eccda4d..13b6cbf 100644
--- a/data/ui/tasksel.glade
+++ b/data/ui/tasksel.glade
@@ -107,7 +107,7 @@
 	    <widget class="GtkLabel" id="label4">
 	      <property name="width_request">600</property>
 	      <property name="visible">True</property>
-	      <property name="label" translatable="yes">Please select any additional repositories that you want to use for software installation.</property>
+	      <property name="label" translatable="yes">Repositories that will be used for software installation.</property>
 	      <property name="use_underline">False</property>
 	      <property name="use_markup">False</property>
 	      <property name="justify">GTK_JUSTIFY_LEFT</property>
diff --git a/pyanaconda/iw/task_gui.py b/pyanaconda/iw/task_gui.py
index 8ff8add..d21f7c3 100644
--- a/pyanaconda/iw/task_gui.py
+++ b/pyanaconda/iw/task_gui.py
@@ -657,6 +657,7 @@ class TaskWindow(InstallWindow):
                               gobject.TYPE_PYOBJECT)
 
         tl = self.xml.get_widget("repoList")
+        tl.set_sensitive(False)
         tl.set_model(store)
 
         cbr = gtk.CellRendererToggle()
@@ -706,7 +707,11 @@ class TaskWindow(InstallWindow):
             self.xml.get_widget("cbVBox").hide()
             self.xml.get_widget("mainLabel").hide()
 
-        self.xml.get_widget("addRepoButton").connect("clicked", self._addRepo)
-        self.xml.get_widget("editRepoButton").connect("clicked", self._editRepo, self.rs)
+        #self.xml.get_widget("addRepoButton").connect("clicked", self._addRepo)
+        #self.xml.get_widget("editRepoButton").connect("clicked", self._editRepo, self.rs)
+        self.xml.get_widget("addRepoButton").hide()
+        self.xml.get_widget("editRepoButton").hide()
+        self.xml.get_widget("addRepoButton").set_no_show_all(True)
+        self.xml.get_widget("editRepoButton").set_no_show_all(True)
 
         return vbox
-- 
1.8.3.1

